File: rest/taskrouter/workers/list/get/example-2/example-2.rb
# Get twilio-ruby from twilio.com/docs/ruby/install
require 'rubygems'          # This line not needed for ruby > 1.8
require 'twilio-ruby'

# Get your Account Sid and Auth Token from twilio.com/user/account
account_sid = '{{ account_sid }}'
auth_token = '{{ auth_token }}'
workspace_sid = '{{ workspace_sid }}'

client = Twilio::REST::TaskRouterClient.new account_sid, auth_token, workspace_sid

client.workspace.workers.list({"Available":"1"}).each do |worker|
  puts worker.friendly_name
end

client.workspace.workers.list({"TaskQueueSid":"WQf855e98ad280d0a0a325628e24ca9627", "Available":"1"}).each do |worker|
  puts worker.friendly_name
end

client.workspace.workers.list({"TargetWorkersExpression":"type == 'leads'"}).each do |worker|
  puts worker.friendly_name
end

File: rest/taskrouter/tasks/list/get/example-2/example-2.js
// Download the Node helper library from twilio.com/docs/node/install
// These vars are your accountSid and authToken from twilio.com/user/account
var twilio = require('twilio');

var accountSid = "{{ account_sid }}";
var authToken = "{{ auth_token }}";
var workspaceSid = "{{ workspace_sid }}";

var client = new twilio.TaskRouterClient(accountSid, authToken, workspaceSid);

client.workspace.tasks.get({"TaskQueueSid":"WQf855e98ad280d0a0a325628e24ca9627"}, function(err, data) {
    if(!err) {
        data.tasks.forEach(function(task) {
            console.log(task.attributes);
        })
    }
});

client.workspace.tasks.get({"AssignmentStatus":"pending"}, function(err, data) {
    if(!err) {
        data.tasks.forEach(function(task) {
            console.log(task.attributes);
        })
    }
});